test/test_changelog.py

```diff
@@ -3,72 +3,112 @@
+        # using this repository as parent folder
         self.assertTrue(ChangelogParser.has_changelog())
-        parser = ChangelogParser(CHANGELOG_REGEXP)
-        self.assertIsNone(parser.content("0.0.0"), "")
+        self.assertIsInstance(ChangelogParser.CHANGELOG_FILEPATH, Path)
 
-        expected = (
-            "* Tag using a wrong format DD/MM/YYYY according to Keep A Changelog\n"
-            '* Tag without "v" prefix\n'
-            "* Add a CHANGELOG.md file for testing"
-        )
-        self.assertEqual(parser.content("0.1.2"), expected)
-
-        expected = (
-            "\n "
-            "Version 0.1.2 :\n "
-            "* Tag using a wrong format DD/MM/YYYY according to Keep A Changelog\n "
-            '* Tag without "v" prefix\n '
-            "* Add a CHANGELOG.md file for testing\n"
-            "\n"
+        # using the fixture subfolder as string
+        self.assertTrue(ChangelogParser.has_changelog(parent_folder="test/fixtures"))
+        self.assertIsInstance(ChangelogParser.CHANGELOG_FILEPATH, Path)
+
+        # using the fixture subfolder as pathlib.Path
+        self.assertTrue(
+            ChangelogParser.has_changelog(parent_folder=Path("test/fixtures"))
         )
-        self.assertEqual(parser.last_items(1), expected)
+        self.assertIsInstance(ChangelogParser.CHANGELOG_FILEPATH, Path)
 
-        expected = """
- Version 0.1.2 :
- * Tag using a wrong format DD/MM/YYYY according to Keep A Changelog
- * Tag without "v" prefix
- * Add a CHANGELOG.md file for testing
+        # with a path to a file, must raise a type error
+        with self.assertRaises(TypeError):
+            ChangelogParser.has_changelog(parent_folder=Path(__file__))
+        self.assertIsNone(ChangelogParser.CHANGELOG_FILEPATH, None)
 
- Version v0.1.1 :
- * Tag using a correct format YYYY-MM-DD according to Keep A Changelog
- * Tag with a "v" prefix to check the regular expression
- * Previous version
+        # with a path to a folder which doesn't exist, must raise a file exists error
+        with self.assertRaises(FileExistsError):
+            ChangelogParser.has_changelog(parent_folder=Path("imaginary_path"))
 
- Version 0.1.0 :
- * Very old version
+    def test_changelog_content(self):
+        """Test version content from changelog."""
+        # parser
+        parser = ChangelogParser(parent_folder="test/fixtures")
+        self.assertIsInstance(parser.CHANGELOG_FILEPATH, Path)
 
-"""
-        self.assertEqual(parser.last_items(3), expected)
+        expected_version_note = (
+            "- This is a version with a prerelease in this changelog\n"
+            "- The changelog module is tested against these lines\n"
+            "- Be careful modifying this file"
+        )
+
+        # checks
+        self.assertIsNone(parser.content("0.0.0"))
+        self.assertIsInstance(parser.content("10.1.0-alpha1"), str)
+        self.assertEqual(parser.content("10.1.0-alpha1"), expected_version_note)
```

Comment:
Would you mind adding all versions in the changelog.md as tests ?

Reply:
Sorry, I don't get your point. What do you mean by "adding all versions"?

Reply:
Checking in the tests, that the parser can read all individual versions. This will cover the regexp used : against different formats, dates, v prefix etc

I just tried to add the test : 
```python
        expected = (
            "* Tag with a \"v\" prefix to check the regular expression\n"
            "* Previous version"
        )
        self.assertEqual(expected, parser.content(tag="v0.1.1"))
```

```
AssertionError: '* Tag with a "v" prefix to check the regular expression\n* Previous version' != None
```

Did I write something wrong ?